s4e-web/src/app/views/settings/components/generic-list-view/generic-list-view.component.ts

```diff
@@ -0,0 +1,74 @@
+import {Component, ContentChild, EventEmitter, Input, OnDestroy, OnInit, Output, TemplateRef} from '@angular/core';
+import {FormArray, FormControl} from '@ng-stack/forms';
+import {untilDestroyed} from 'ngx-take-until-destroy';
+import {ID} from '@datorama/akita';
+import {Subscription} from 'rxjs';
+import {debounceTime} from 'rxjs/operators';
+
+@Component({
+  selector: 's4e-generic-list-view',
+  templateUrl: './generic-list-view.component.html',
+  styleUrls: ['./generic-list-view.component.scss']
+})
+export class GenericListViewComponent implements OnInit, OnDestroy {
+  @ContentChild(TemplateRef) tableRow: TemplateRef<any>;
+  @Input() error: any|null = null;
+  @Input() loading: boolean = true;
+  @Input() itemIdFunction: (any) => ID = (item: any) => item.id;
+  @Input() itemSearchFunction: (any, query) => boolean = (item: any, query: string) => Object.values(item)
+    .filter(v => typeof v == 'string').map((v: string) => v.toLocaleLowerCase()).join(';').indexOf(query.toLocaleLowerCase()) >= 0;
```

Comment:
Could you change the `v` name to more meaningfull, like `item` or `value`?

Reply:
`v` usually means `value`. The same way as `i` in `for loops` means `index`. But I won't argue here if it's more readable for you then no problem